targets/TARGET_GigaDevice/TARGET_GD32F30X/mbed_overrides.c

```diff
@@ -0,0 +1,154 @@
+    } while ((0U == stab_flag) && (HXTAL_STARTUP_TIMEOUT != timeout));
+
+    /* if fail */
+    if (0U == (RCU_CTL & RCU_CTL_HXTALSTB)) {
+        while (1) {
+        }
+    }
+
+    RCU_APB1EN |= RCU_APB1EN_PMUEN;
+    PMU_CTL |= PMU_CTL_LDOVS;
+
+    /* HXTAL is stable */
+    /* AHB = SYSCLK */
+    RCU_CFG0 |= RCU_AHB_CKSYS_DIV1;
+    /* APB2 = AHB/1 */
+    RCU_CFG0 |= RCU_APB2_CKAHB_DIV1;
+    /* APB1 = AHB/2 */
+    RCU_CFG0 |= RCU_APB1_CKAHB_DIV2;
+
+#if (defined(GD32F30X_HD) || defined(GD32F30X_XD))
+    /* select HXTAL/2 as clock source */
+    RCU_CFG0 &= ~(RCU_CFG0_PLLSEL | RCU_CFG0_PREDV0);
+    RCU_CFG0 |= (RCU_PLLSRC_HXTAL_IRC48M | RCU_CFG0_PREDV0);
+
+    /* CK_PLL = (CK_HXTAL/2) * 30 = 120 MHz */
+    RCU_CFG0 &= ~(RCU_CFG0_PLLMF | RCU_CFG0_PLLMF_4 | RCU_CFG0_PLLMF_5);
+    RCU_CFG0 |= RCU_PLL_MUL30;
+
+#elif defined(GD32F30X_CL)
+    /* CK_PLL = (CK_PREDIV0) * 30 = 120 MHz */
+    RCU_CFG0 &= ~(RCU_CFG0_PLLMF | RCU_CFG0_PLLMF_4 | RCU_CFG0_PLLMF_5);
+    RCU_CFG0 |= (RCU_PLLSRC_HXTAL_IRC48M | RCU_PLL_MUL30);
+
+    /* CK_PREDIV0 = (CK_HXTAL)/5 *8 /10 = 4 MHz */
+    RCU_CFG1 &= ~(RCU_CFG1_PLLPRESEL | RCU_CFG1_PREDV0SEL | RCU_CFG1_PLL1MF | RCU_CFG1_PREDV1 | RCU_CFG1_PREDV0);
+    RCU_CFG1 |= (RCU_PLLPRESRC_HXTAL | RCU_PREDV0SRC_CKPLL1 | RCU_PLL1_MUL8 | RCU_PREDV1_DIV5 | RCU_PREDV0_DIV10);
+
+    /* enable PLL1 */
+    RCU_CTL |= RCU_CTL_PLL1EN;
+    /* wait till PLL1 is ready */
+    while ((RCU_CTL & RCU_CTL_PLL1STB) == 0U) {
+    }
+#endif /* GD32F30X_HD and GD32F30X_XD */
+
+    /* enable PLL */
+    RCU_CTL |= RCU_CTL_PLLEN;
+
+    /* wait until PLL is stable */
+    while (0U == (RCU_CTL & RCU_CTL_PLLSTB)) {
+    }
+
+    /* enable the high-drive to extend the clock frequency to 120 MHz */
+    PMU_CTL |= PMU_CTL_HDEN;
+    while (0U == (PMU_CS & PMU_CS_HDRF)) {
+    }
+
+    /* select the high-drive mode */
+    PMU_CTL |= PMU_CTL_HDS;
+    while (0U == (PMU_CS & PMU_CS_HDSRF)) {
+    }
+
+    /* select PLL as system clock */
+    RCU_CFG0 &= ~RCU_CFG0_SCS;
+    RCU_CFG0 |= RCU_CKSYSSRC_PLL;
+
+    /* wait until PLL is selected as system clock */
+    while (0U == (RCU_CFG0 & RCU_SCSS_PLL)) {
+    }
+}
+
+/**
+ * SDK hook for running code before ctors or OS
+ *
+ * This is a weak function which can be overridden by a target's
+ * SDK to allow code to run after ram is initialized but before
+ * the OS has been started or constructors have run.
+ *
+ * Preconditions:
+ * - Ram is initialized
+ * - NVIC is setup
+ */
+/**
+ * This function is called after RAM initialization and before main.
+ */
+void mbed_sdk_init()
+{
+    /* Update the SystemCoreClock */
+    SystemCoreClockUpdate();
+    nvic_priority_group_set(NVIC_PRIGROUP_PRE4_SUB0);
+
+    /* configure 1ms tick */
+#if TICKER_TIMER_WIDTH_BIT == 16
+    ticker_16bits_timer_init();
+#else
+    ticker_32bits_timer_init();
+#endif
+
+    system_clock_120m_hxtal();
+
+    SystemCoreClockUpdate();
+
+    mbed_sdk_inited = 1;

```

Comment:
This access does not need to be thread/interupt safe (this is used in the HAL as I've seen)

Reply:
‘mbed_sdk_inited’ is used in our firmware library for obtaining hardware delay at startup. It's no problem to delete it, but it is not conducive to our later expansion. 

So the question is, SHOULD it be deleted from our firmware library? Thanks a lot. @0xc0170

Reply:
It can stay, I was just curious is access like this is fine (no race conditions that might lead do undefined behavior)

Reply:
All good.